syntax = "proto3";

package xray_config;

service XrayConfigService {
  rpc CreateClient (CreateClientRequest) returns (OperationResponse);
  rpc ReadClient (ReadClientRequest) returns (Client);
  rpc UpdateClient (UpdateClientRequest) returns (OperationResponse);
  rpc DeleteClient (DeleteClientRequest) returns (OperationResponse);
}

message Client {
  string password = 1;
  string method = 2;
  string email = 3;
}

message CreateClientRequest {
  Client client = 1;
}

message ReadClientRequest {
  string email = 1;
}

message UpdateClientRequest {
  string email = 1;
  Client client = 2;
}

message DeleteClientRequest {
  string email = 1;
}

message OperationResponse {
  bool success = 1;
  string message = 2;
}